### PR TITLE
Issue 132 add changeset printout

### PR DESCRIPTION
**Pull request by tparchambault**
The state manager maintains a list of applied but not deployed changes. This PR allows the state manager to request a text based list of path/operation pairs from the lower level rust fapolicyd interface layer, which can then be used by the UI layers.

### CODE REVIEW

File: python/fapolicy_analyzer/ui/state_manager.py [hunk 1]
```diff
@@ -73,5 +73,16 @@ def __update_dirty_queue(self):
...
+    def get_path_action_list(self):
+        listReturn = []
+
+        # Iterate through the StateManagers Changeset list
+        for e in self.listChangeset:
+            # Each changeset contains a dict with at least one Path/Action pair
+            for t in e.get_path_action_map().items():
+                listReturn.append(t)
+
+        return listReturn
```
Comment (dorschs57) [inline python/fapolicy_analyzer/ui/state_manager.py:85]: This whole think can be done with python comprehensions 
```suggestion
        return [t for e in self.listChangeset for t in e.get_path_action_map().items]
```

File: src/trust.rs [hunk 2]
```diff
@@ -156,6 +156,14 @@ impl TrustOp {
             }
         }
     }
+
+    fn to_string(&self) -> (String, String) {
```
Comment (jw3) [inline src/trust.rs:160]: lets rename this to something like `to_pair` and make it a function in the module instead of a member

Comment (tparchambault) [reply src/trust.rs:160]: How about 'to_string_pair' I'm a big fan of Obvious.

File: python/fapolicy_analyzer/ui/state_manager.py (hunk 1 above)
Comment (tparchambault) [reply python/fapolicy_analyzer/ui/state_manager.py]: Done

File: src/trust.rs (hunk 2 above)
Comment (tparchambault) [reply src/trust.rs]: to_pair( &TrustOp). Done.

File: src/trust.rs [hunk 3]
```diff
@@ -197,6 +205,14 @@ impl Changeset {
     pub fn is_empty(&self) -> bool {
         self.changes.is_empty()
     }
+
+    pub fn get_path_action_map(&self) -> HashMap<String, String> {
+        let mut map_path_action = HashMap::new();
```
[the remark this answers is not in the document]
Comment (tparchambault) [reply src/trust.rs]: More or less, as suggested. Done.

File: src/trust.rs [hunk 4]
```diff
@@ -197,6 +205,14 @@ impl Changeset {
     pub fn is_empty(&self) -> bool {
         self.changes.is_empty()
     }
+
+    pub fn get_path_action_map(&self) -> HashMap<String, String> {
```
[the remark this answers is not in the document]
Comment (tparchambault) [reply src/trust.rs]: The two functions in src/trust.rs have been moved to the module level.

**Review by jw3 — APPROVED** [review]
Looks like it needs a `cargo fmt --all`, otherwise all good here :+1:

**Comment by dorschs57** [comment]
Good here too 👍

**Review by dorschs57 — APPROVED** [review]

**Pull request merged**